### PR TITLE
API changes for 4.1

### PR DESCRIPTION
**Pull request by NolaDonato**
These API changes are up for review. Corresponding pull requests in GearVRf-Demos and GearVRf-Tests incorporate these api changes.

Removed GVRMaterialShaderManager, GVRPostEffectShaderManager (use GVRShaderManager now)
Removed GVRGUISceneObject (GVRViewSceneObject has this functionality now)
GVRBaseSensor -> GVRSensor
GVRBitmapTexture -> GVRBitmapImage
GVRCubemapTexture -> GVRCubemapImae
GVRCompressedTexture -> GVRCompressedImage
GVRCompressedCubemapTexture -> GVRCompressedCubemapImage
GVRFloatTexture -> GVRFloatImage
Removed GVRMesh.setTriangles, GVRMesh.getTriangles
Added GVRMesh.setCharIndices, GVRMesh.setIntIndices, GVRMesh.getCharIndices, GVRMesh.getIntIndices
GVRLightBase -> GVRLight
GearCursorController -> GVRGearCursorController
Moved GVRxxxShader to org.gearvrf.shaders
Move GVRCursorController to org.gearvrf.io

GearVRF DCO signed off by: Nola Donato nola.donato@samsung.com

### CODE REVIEW

**Comment by liaxim** [comment]
Have not examined the change but in general: new packages are only ok if the dependencies are one-way (to org.gearvrf), not too many (hard to set exact number) and there is a clean API boundary (no make this and that public just to get to it).

**Comment by liaxim** [comment]
Needs rebase

**Comment by NolaDonato** [comment]
It needs internal discussion more than rebase :-)

File: GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRCursorController.java [hunk 1]
```diff
@@ -988,7 +1002,7 @@ public boolean isEventHandledBySensorManager() {
         return eventHandledBySensor;
     }
 
-    void setEventHandledBySensor()
+    public void setEventHandledBySensor()
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRCursorController.java]: setEventHandledBySensor is an example why I am against having the io package. As long as we need such methods to be public, we should keep the io stuff in the main package.

**Comment by liaxim** [comment]
It is a nice improvement. You should rebase so we can get it in. Please consider the need for the io package but I am not going to get hung up on that.

**Comment by thomasflynn** [comment]
@liaxim are there any other API changes you would like to see?  (other than the io package that you mentioned)

**Comment by liaxim** [comment]
Have nothing specific. But it would be nice to remove at least some of the deprecated methods.

**Comment by liaxim** [comment]
Well, I find ``public void addChildObject(GVRComponent childComponent)`` not particularly good. Should be removed in my opinion and its use appropriately replaced with ``addChildObject(GVRSceneObject child)``.

**Comment by NolaDonato** [comment]
I agree about GVRComponent.addChildObject. I left that in for compatibility with old stuff but it can be removed.

**Comment by NolaDonato** [comment]
If there are no complaints I will merge this tomorrow. Tests won't run without it.

**Comment by liaxim** [comment]
Needs rebase

**Pull request merged**